wsmaster/che-core-api-factory-github-common/src/main/java/org/eclipse/che/api/factory/server/github/AbstractGithubURLParser.java

```diff
@@ -70,26 +75,77 @@ public abstract class AbstractGithubURLParser {
     this.tokenManager = tokenManager;
     this.devfileFilenamesProvider = devfileFilenamesProvider;
     this.apiClient = githubApiClient;
-    this.oauthEndpoint = oauthEndpoint;
     this.disableSubdomainIsolation = disableSubdomainIsolation;
     this.providerName = providerName;
 
     String endpoint =
         isNullOrEmpty(oauthEndpoint) ? GITHUB_SAAS_ENDPOINT : trimEnd(oauthEndpoint, '/');
 
-    this.githubPattern =
-        compile(
-            format(
-                "^%s/(?<repoUser>[^/]+)/(?<repoName>[^/]++)((/)|(?:/tree/(?<branchName>.++))|(/pull/(?<pullRequestId>\\d++)))?$",
-                endpoint));
+    this.githubPattern = compile(format(githubPatternTemplate, endpoint));
     this.githubSSHPattern =
-        compile(format("^git@%s:(?<repoUser>.*)/(?<repoName>.*)$", URI.create(endpoint).getHost()));
+        compile(format(githubSSHPatternTemplate, URI.create(endpoint).getHost()));
   }
 
   public boolean isValid(@NotNull String url) {
     String trimmedUrl = trimEnd(url, '/');
     return githubPattern.matcher(trimmedUrl).matches()
-        || githubSSHPattern.matcher(trimmedUrl).matches();
+        || githubSSHPattern.matcher(trimmedUrl).matches()
+        || isUserTokenPresent(trimmedUrl)
+        || isApiRequestRelevant(trimmedUrl);
+  }
```

Comment:
Why do we need extra checks here?

Reply:
The url might correspond to a GitHub server which is not configured so the url is unpredictable. We make a test request in case if the project is public so we could use it even without a PAT. I'll add some descriptions here.